packages/web-components/src/components/footer/language-selector.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020, 2021
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html, property, customElement, query, internalProperty } from 'lit-element';
+import ddsSettings from '@carbon/ibmdotcom-utilities/es/utilities/settings/settings.js';
+import BXComboBox, { DROPDOWN_SIZE } from 'carbon-web-components/es/components/combo-box/combo-box.js';
+import BXComboBoxItem from 'carbon-web-components/es/components/combo-box/combo-box-item.js';
+import HostListener from 'carbon-web-components/es/globals/decorators/host-listener.js';
+import HostListenerMixin from 'carbon-web-components/es/globals/mixins/host-listener.js';
+import { LANGUAGE_SELECTOR_STYLE_SCHEME } from './defs';
+import styles from './footer.scss';
+import { findIndex, forEach } from '../../globals/internal/collection-helpers';
+
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * Language selector component.
+ * The API for language selection is still subject to change.
+ *
+ * @element dds-language-selector
+ * @internal
+ */
+@customElement(`${ddsPrefix}-language-selector`)
+class DDSLanguageSelector extends HostListenerMixin(BXComboBox) {
+  /**
+   * Property that saves the last valid language to use on reset cases.
+   */
+
+  @internalProperty()
+  _lastValidLang?: string;
+
+  @property()
+  size = DROPDOWN_SIZE.EXTRA_LARGE;
+
+  /**
+   * Size property to apply different styles.
+   */
+
+  @property({ attribute: 'footer-size' })
+  footerSize = LANGUAGE_SELECTOR_STYLE_SCHEME.REGULAR;
+
+  /**
+   * The `<input>` for filtering.
+   */
+
+  @query('input')
+  private filterInputNode!: HTMLInputElement;
```

Comment:
Please make sure private properties/methods are prefixed with `_`:

```suggestion
  private _filterInputNode!: HTMLInputElement;
```